alert-database/src/main/java/com/synopsys/integration/alert/database/api/DefaultProviderDataAccessor.java

```diff
@@ -58,56 +54,52 @@
-    public static final Integer DEFAULT_LIMIT = 100;
-
     public static final int MAX_DESCRIPTION_LENGTH = 250;
     public static final int MAX_PROJECT_NAME_LENGTH = 507;
 
     private final Logger logger = LoggerFactory.getLogger(DefaultProviderDataAccessor.class);
     private final ProviderProjectRepository providerProjectRepository;
     private final ProviderUserProjectRelationRepository providerUserProjectRelationRepository;
     private final ProviderUserRepository providerUserRepository;
+    private final ConfigurationAccessor configurationAccessor;
 
     @Autowired
-    public DefaultProviderDataAccessor(ProviderProjectRepository providerProjectRepository, ProviderUserProjectRelationRepository providerUserProjectRelationRepository, ProviderUserRepository providerUserRepository) {
+    public DefaultProviderDataAccessor(ProviderProjectRepository providerProjectRepository, ProviderUserProjectRelationRepository providerUserProjectRelationRepository, ProviderUserRepository providerUserRepository,
+        ConfigurationAccessor configurationAccessor) {
         this.providerProjectRepository = providerProjectRepository;
         this.providerUserProjectRelationRepository = providerUserProjectRelationRepository;
         this.providerUserRepository = providerUserRepository;
+        this.configurationAccessor = configurationAccessor;
     }
 
     @Override
     @Transactional(readOnly = true, isolation = Isolation.READ_COMMITTED)
-    public Optional<ProviderProject> findFirstByHref(String href) {
-        return providerProjectRepository.findFirstByHref(href).map(this::convertToProjectModel);
-    }
-
-    @Override
-    @Transactional(readOnly = true, isolation = Isolation.READ_COMMITTED)
-    public Optional<ProviderProject> findFirstByName(String name) {
-        return providerProjectRepository.findFirstByName(name).map(this::convertToProjectModel);
-    }
-
-    @Override
-    @Transactional(readOnly = true, isolation = Isolation.READ_COMMITTED)
-    public List<ProviderProject> findByProviderName(String providerName) {
-        return providerProjectRepository.findByProvider(providerName)
-                   .stream()
-                   .map(this::convertToProjectModel)
-                   .collect(Collectors.toList());
+    public List<ProviderProject> findByProviderConfigName(String providerConfigName) {
+        try {
+            Optional<Long> optionalConfigId = configurationAccessor.getProviderConfigurationByName(providerConfigName)
+                                                  .map(ConfigurationModel::getConfigurationId);
+            if (optionalConfigId.isPresent()) {
+                return providerProjectRepository.findByProviderConfigId(optionalConfigId.get())
+                           .stream()
+                           .map(this::convertToProjectModel)
+                           .collect(Collectors.toList());
+            }
+        } catch (AlertDatabaseConstraintException ignored) {
```

Comment:
Why are we ignoring this exception?

Reply:
If the configId is present, then the database constraint is fulfilled, meaning that specific exception will not be thrown. Because this method is more of a "search for provider configs by name", maybe it would be better to return an `Optional` than a `List`.

Reply:
Not to mention that because this is a search for projects, we don't necessarily want to throw an exception, but instead return an empty `List`.

Reply:
You may want to add a comment to avoid future confusion. But given your explanation this seems fine